src/languages/fsharp.js

```diff
@@ -154,15 +408,37 @@ export default function(hljs) {
         ]
       },
       {
-        scope: 'operator',
         // only non arithmetic operators that we can confidently match:
-        match: /\|{1,3}>|<\|{1,3}|->|<-|(?<!<)<<(?!<)|(?<!>)>>(?!>)|:>|:\?>?|\.\.|::|(?<!\|)\|(?!\|)/
+        scope: 'operator',
+        // split in two because we don't want the same relevance for all operators.
+        // warning: the order of the two modes is important, because |> collides with |
+        variants: [
+          {
+            // fsharp specific operators:
+            match: regex.either(
+              /\|{1,3}>/, // |> ||> |||>
+              /<\|{1,3}/, // <| <|| <|||
+              /(?<!<)<<(?!<)/, // << (but not < or <<<)
+              /(?<!>)>>(?!>)/, // >> (but not > or >>>)
```

Comment:
We don't yet support negative lookbehind because not all browsers support it well yet.

Reply:
Hmm, honestly I don't know what to do about that. What do you suggest?

- Match the variants I didn't want to match and accept having no real logic to what is highlighted or not? (`>>` is used to combine functions, `>>>` is a binary right shift)
- Do some magic I don't know about to replace the negative look behind?
- Do some hack by matching space before/after?
- ...?

(Do you know how it behaves on browsers that don't support it? Can't we keep it if it fails more or less gracefully)

Reply:
> If you don't want to highlight >>> just match it without a scope. 

Ah that makes sense! I guess it didn't occur to me it was an option :)

> Though why are we highlighting some operators and not others?

Mostly because I am afraid to highlight things in places where they should not be highlighted.
[F# allows to define new operators, using sequences of certain characters](https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/operator-overloading), and I didn't want to risk only partially highlighting a custom operator for example. So I thought sticking with standard non-arithmetic operators would be a good compromise for now.

Reply:
Hmmm, now that I'm looking closer not highlighting the basic math operators seems very strange to me... couldn't we just detect "operator like expressions" with a regex and just highlight ALL potential custom defined operators?  I'm not sure that would have been my first choice, but I really dislike not highlighting the basic operators.

Would doing this conflict with other language features we're not already parsing?

Reply:
> Would doing this conflict with other language features we're not already parsing?

That's part of what worries me... For example I think I removed explicit parsing of the generic type definition brackets to simplify things. They would need to be re-added, along with some more matching for this pattern when it's not part of a type definition (e.g. `let f (x: inref<SomeStruct>) = x.SomeField`)

I fear there are a lot of patterns like that I don't necessarily have in mind...

Reply:
I initially wanted to highlight only the most idiomatic operators (maybe `|>`, `<|`, `>>`, `<<`), but it seemed too arbitrary, that's why I settled on the set I chose...

Reply:
Well... now that you mention it, `<>` is the boolean not operator in F# ^^'

It's a bit sad, but maybe not matching any operator is safer then...

It's really frustrating, especially after I thought I managed to do what I wanted with look-behinds...